modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1022,7 +1023,21 @@ public final HitInfo hitTest(Point2D point) {
         TextLayout layout = getTextLayout();
         double x = point.getX() - getX();
         double y = point.getY() - getY() + getYRendering();
-        TextLayout.Hit h = layout.getHitInfo((float)x, (float)y, getText());
+        GlyphList[] runs = getRuns();
+        int runIndex = 0;
+        if (runs.length != 0) {
+            Point2D ptInParent = localToParent(x, y);
+            while (runIndex < runs.length - 1) {
+                if (ptInParent.getY() > runs[runIndex].getLocation().y
+                        && ptInParent.getY() < runs[runIndex + 1].getLocation().y) {
```

Comment:
[minor]: the formatting looks off.  If we rename 'ptInParent` to `p` it will fit on one line.
Also, ptInParent.getY() gets called twice, may be use a local variable?  Also will make it shorter.